api/interactives.go

```diff
@@ -198,17 +192,18 @@ func (api *API) UpdateInteractiveHandler(w http.ResponseWriter, req *http.Reques
 
 	// 4. write to DB
 	err = api.mongoDB.UpsertInteractive(ctx, id, &models.Interactive{
-		MetadataJson: string(mDataJson),
-		State:        state.String(),
-		Archive:      archive,
+		Metadata: update.Interactive.Metadata,
+		State:    state.String(),
+		Archive:  archive,
 	})
 	if err != nil {
 		http.Error(w, err.Error(), http.StatusInternalServerError)
 		log.Error(ctx, "Unable to write to DB", err)
 		return
 	}
 
-	WriteJSONBody(mergedMap, w, http.StatusOK)
+	WriteJSONBody(update.Interactive.Metadata, w, http.StatusOK)
```

Comment:
shall we return the `interactive` here? for all responses too - be consistent for them all. controller will need access to files bit. more restful too - although we are not strictly rest. 😊